src/semidiscretization/semidiscretization.jl

```diff
@@ -302,6 +302,12 @@ function wrap_array_native(u_ode, semi::AbstractSemidiscretization)
 end
 
 
+function summary_solver(semi::AbstractSemidiscretization)
```

Comment:
Why not 
```suggestion
function summary(semi::AbstractSemidiscretization)
```
and let dispatch do the rest?

Reply:
For some reason, this turned out to be harder than expected. Just changing the name of the function caused the call two lines below (`summary(solver)`) to fail. It suddenly can't find the function `summary(DGSEM)`. Changing this line to `Base.summary(solver)` works for some reason, but then it fails for the FDSBP elixir. We're a bit confused and out of ideas now.

Reply:
@ranocha Do you see what the problem is here?

Reply:
`Base.summary`?

Reply:
Do you mean changing `summary(solver)` to `Base.summary(solver)`?
Then, we get
```
LoadError: MethodError: no method matching summary(::SummationByPartsOperators.DerivativeOperator{Float64, Tuple{SummationByPartsOperators.DerivativeCoefficientRow{Float64, 1, 4}, SummationByPartsOperators.DerivativeCoefficientRow{Float64, 1, 3}, SummationByPartsOperators.DerivativeCoefficientRow{Float64, 1, 5}, SummationByPartsOperators.DerivativeCoefficientRow{Float64, 1, 6}}, Tuple{SummationByPartsOperators.DerivativeCoefficientRow{Float64, 1, 4}, SummationByPartsOperators.DerivativeCoefficientRow{Float64, 1, 3}, SummationByPartsOperators.DerivativeCoefficientRow{Float64, 1, 5}, SummationByPartsOperators.DerivativeCoefficientRow{Float64, 1, 6}}, Tuple{}, Tuple{}, 2, 2, 4, 4, SummationByPartsOperators.FastMode, SummationByPartsOperators.MattssonNordström2004, StepRangeLen{Float64, Base.TwicePrecision{Float64}, Base.TwicePrecision{Float64}}})
  You may have intended to import Base.summary
  Closest candidates are:
    summary(::Trixi.SemidiscretizationCoupled) at /home/erik/git/Trixi.jl/src/semidiscretization/semidiscretization_coupled.jl:85
    summary(::Trixi.AbstractSemidiscretization) at /home/erik/git/Trixi.jl/src/semidiscretization/semidiscretization.jl:305
  Stacktrace:
    [1] summary(io::IOBuffer, dg::Trixi.DG{SummationByPartsOperators.DerivativeOperator{Float64, Tuple{SummationByPartsOperators.DerivativeCoefficientRow{Float64, 1, 4}, SummationByPartsOperators.DerivativeCoefficientRow{Float64, 1, 3}, SummationByPartsOperators.DerivativeCoefficientRow{Float64, 1, 5}, SummationByPartsOperators.DerivativeCoefficientRow{Float64, 1, 6}}, Tuple{SummationByPartsOperators.DerivativeCoefficientRow{Float64, 1, 4}, SummationByPartsOperators.DerivativeCoefficientRow{Float64, 1, 3}, SummationByPartsOperators.DerivativeCoefficientRow{Float64, 1, 5}, SummationByPartsOperators.DerivativeCoefficientRow{Float64, 1, 6}}, Tuple{}, Tuple{}, 2, 2, 4, 4, SummationByPartsOperators.FastMode, SummationByPartsOperators.MattssonNordström2004, StepRangeLen{Float64, Base.TwicePrecision{Float64}, Base.TwicePrecision{Float64}}}, Nothing, Trixi.SurfaceIntegralStrongForm{Trixi.FluxLaxFriedrichs{typeof(Trixi.max_abs_speed_naive)}}, Trixi.VolumeIntegralStrongForm})
      @ Trixi ~/git/Trixi.jl/src/solvers/dg.jl:260
[...]
```

Reply:
It works when we also replace `summary(dg.basis)` with `Base.summary(dg.basis)` in `solvers/dg.jl`, but is this the correct fix? Why do we even have to change something in this file?

Reply:
If you define `function summary(semi::AbstractSemidiscretization)`, you create a new function `summary` within Trixi.jl. This new function shadows `Base.summary`, which is what we want to call there. Thus, whenever you want to define a new method of `summary` extending `Base.summary`, you need to use the prefix `Base.`.

Reply:
Thus, the question is: Where do you define new methods of `summary` in this PR?

Reply:
Okay, now I see what's happening here.
We introduced this function `summary(AbstractSemidiscretization)` and `summary(SemidiscretizationCoupled)`. Now, calling `summary(basis)` here
https://github.com/trixi-framework/Trixi.jl/blob/9ee86ebb95dbcbf90fca6e4bea011567d06c44f9/src/solvers/dg.jl#L260
stopped working because `Base.summary` is now shadowed. It works if I replace this by `Base.summary(dg.basis)`.

Should we do it that way? Or should we keep the name `summary_solver` here, not following @sloede's suggestion, to not shadow `Base.summary`?

Reply:
Why don't you just make 

> `summary(AbstractSemidiscretization)` and `summary(SemidiscretizationCoupled)`

methods of `Base.summary`?

Reply:
That works out, thanks!